livekit-api/livekit/api/sip_service.py

```diff
@@ -1,33 +1,77 @@
 import aiohttp
+import re
+from typing import Optional
+
+from livekit.protocol.models import ListUpdate
 from livekit.protocol.sip import (
-    CreateSIPTrunkRequest,
     SIPTrunkInfo,
     CreateSIPInboundTrunkRequest,
+    UpdateSIPInboundTrunkRequest,
     SIPInboundTrunkInfo,
+    SIPInboundTrunkUpdate,
     CreateSIPOutboundTrunkRequest,
+    UpdateSIPOutboundTrunkRequest,
     SIPOutboundTrunkInfo,
-    ListSIPTrunkRequest,
-    ListSIPTrunkResponse,
+    SIPOutboundTrunkUpdate,
     ListSIPInboundTrunkRequest,
     ListSIPInboundTrunkResponse,
     ListSIPOutboundTrunkRequest,
     ListSIPOutboundTrunkResponse,
     DeleteSIPTrunkRequest,
+    SIPDispatchRule,
     SIPDispatchRuleInfo,
+    SIPDispatchRuleUpdate,
     CreateSIPDispatchRuleRequest,
+    UpdateSIPDispatchRuleRequest,
     ListSIPDispatchRuleRequest,
     ListSIPDispatchRuleResponse,
     DeleteSIPDispatchRuleRequest,
     CreateSIPParticipantRequest,
     TransferSIPParticipantRequest,
     SIPParticipantInfo,
+    SIPTransport,
 )
 from ._service import Service
+from .twirp_client import TwirpError
 from .access_token import VideoGrants, SIPGrants
 
 SVC = "SIP"
 """@private"""
 
+_sip_status_pattern = re.compile(r"sip status: (\d+) \((.*?)\)")
```

Comment:
is this guaranteed?

Reply:
i've changed how we are handling this, we are passing back more consistent errors from the server now